openfe/protocols/openmm_afe/base.py

```diff
@@ -841,19 +902,40 @@ def run(self, dry=False, verbose=True,
         # 2. Get settings
         settings = self._handle_settings()
 
-        # 3. Get system generator
-        system_generator = self._get_system_generator(settings, solv_comp)
-
-        # 4. Get modeller
-        system_modeller, comp_resids = self._get_modeller(
-            prot_comp, solv_comp, smc_comps, system_generator,
-            settings['solvation_settings']
-        )
+        if (
```

Comment:
(just in case this moves towards merge in the next couple of weeks): it may be better if this logic went into the getter than in the `run` of base - the idea here is that `run` should be kept pretty lean so it's easy to swap out class methods.

See https://github.com/OpenFreeEnergy/openfe/blob/ahfe-packmol/openfe/protocols/openmm_afe/base.py#L433 for example.

Reply:
I see the benefit of that design; it didn't come about naturally since the Interchange code path does not always need to call `_get_system_generator`